src/utils/c_utils/src/string_c_utils.c

```diff
@@ -422,50 +422,50 @@ bool MallocAndSubstr(char** target, char* source, size_t len)
  * @param src The source string to be copied.
  * @param destByteLen The size of the destination buffer in bytes, including plus 1 for the null terminator.
```

Comment:
confusing. does one pass in "sizeof(dest)+1" or are you stating that the buffer size should be at least one more than the desired length?

Reply:
That is confusing in retrospect. Was trying to convey that destByteLen should include capacity for the additional nul-term. Will fix in next PR.